app/src/test/kotlin/org/spekframework/speksample/RepositoryTest.kt

```diff
@@ -19,29 +20,30 @@ object RepositoryTest : Spek({
     val apiClient: ApiClient = mockk()
     val call: Call<UserDto> = mockk()
     val userDto: UserDto = mockk()
-    val resultDto: Result<UserDto> = mockk()
 
     val repository: Repository = Repository(apiClient)
     val expectedResult = User("name")
 
-    on("login"){
+    on("login") {
 
         it("should return user object") {
 
-            staticMockk("org.spekframework.speksample.ApiClientKt").use {
+            staticMockk("org.spekframework.speksample.ApiClientKt",
+                    "org.spekframework.speksample.RepositoryKt").use {
+
                 every { apiClient.getUser(any()) } returns call
-                every {
-                    runBlocking {
-                        call.awaitResult()
-                    }
+
+                coEvery {
+                    call.awaitResult()
                 } returns Result.Successful(UserDto("name"))
 
                 every {
                     userDto.toDomain()
                 } returns User("name")
 
                 every {
-                    resultDto.map { it.toDomain() }
+                    Result.Successful(UserDto("name"))
+                            .map(invoke<(UserDto) -> User, User, UserDto>(userDto))
```

Comment:
I would agree this hairy construct requires deep understanding. Basically it's lambda invocation that happens when call is happening. Because compiler is quite silly to deduce datatypes we need this direct specification